include/unifex/unstoppable.hpp

```diff
@@ -22,16 +22,54 @@
 
 namespace unifex {
 namespace _unstoppable {
+
+template <typename Sender>
+struct _sender final {
+  struct type;
+};
+
+template <typename Sender>
+using sender = typename _sender<remove_cvref_t<Sender>>::type;
+
+template <typename Sender>
+struct _sender<Sender>::type final {
+  UNIFEX_NO_UNIQUE_ADDRESS Sender sender_;
+
+  template <
+      template <typename...>
+      class Variant,
+      template <typename...>
+      class Tuple>
+  using value_types = sender_value_types_t<Sender, Variant, Tuple>;
+
+  template <template <typename...> class Variant>
+  using error_types = sender_error_types_t<Sender, Variant>;
+
+  static constexpr bool sends_done = sender_traits<Sender>::sends_done;
+
+  template <typename Receiver>
+  friend auto tag_invoke(tag_t<connect>, type self, Receiver&& r) noexcept(
+      is_nothrow_connectable_v<Sender, remove_cvref_t<Receiver>>) {
+    return connect(
+        with_query_value(
+            std::move(self).sender_, get_stop_token, unstoppable_token{}),
```

Comment:
I think this should probably be more constrained, and we should probably take `self` by forwarding reference.
```suggestion
  template(typename Self, typename Receiver) //
      (requires same_as<type, remove_cvref_t<Self>> AND sender_to<member_t<Self, Sender>, Receiver>) //
  friend auto tag_invoke(tag_t<connect>, Self&& self, Receiver&& r) noexcept(
      is_nothrow_connectable_v<member_t<Self, Sender>, remove_cvref_t<Receiver>>) {
    return connect(
        with_query_value(
            static_cast<Self&&>(self).sender_, get_stop_token, unstoppable_token{}),
```